### PR TITLE
andriy.safronov

### PR DESCRIPTION
**Pull request by AndSaf1**

### CODE REVIEW

File: andriy.safronov/03.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+class Tamagochi
+    def status 
+        "Поїсти: #{@eat_time}, Попити: #{@drink_time}, Спати: #{@sleep_time}"
+    end 
+    def initialize
+        @eat_time = 8
+        @drink_time = 6
+        @sleep_time = 5
+    end
```
Comment (alexgonor) [inline andriy.safronov/03.rb]: there should be empty line between methods

File: andriy.safronov/03.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+class Tamagochi
+    def status 
+        "Поїсти: #{@eat_time}, Попити: #{@drink_time}, Спати: #{@sleep_time}"
```
Comment (alexgonor) [inline andriy.safronov/03.rb]: please, write only in english

File: andriy.safronov/03.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    end 
+    def initialize
+        @eat_time = 8
```
Comment (alexgonor) [inline andriy.safronov/03.rb]: indent should be two spaces. You can read this ruby style guide https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md

File: andriy.safronov/03.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    end
+
+   def sleep
+    @sleep_time += 1
+    @eat_time -= 1
+    "Спати"   
+    end
+end
+
+me = Tamagochi.new
```
Comment (alexgonor) [inline andriy.safronov/03.rb]: variable naming confuse, I think maybe you need to create some kind of animal, but not you as an object of tamagochi

File: andriy.safronov/03.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    end
+
+    def drink
+        @drink_time += 1
+        @sleep_time -=1
+        "Попити"
+    end
+
+   def sleep
+    @sleep_time += 1
```
Comment (alexgonor) [inline andriy.safronov/03.rb:25]: please, adjust spaces and indents all over this class

**Review by alexgonor — CHANGES_REQUESTED** [review]

**Review by alexgonor — APPROVED** [review]

**Comment by alexgonor** [comment]
Also, you need to do the second task with brackets validation
Задача 2 Є чотири види дужок - ‘{’ ‘[’ ‘(’ ‘<’ Для кожної з них є відповідна закриваюча дужка - ‘}’ ‘]’ ‘)’ ‘>’
Розширити класс String методом valid? для перевірки строки яка складається лише з дужок на валідність
“{(})“.valid? => false
“{{[]“.valid? => false
“{[()]}“.valid? => true

**Comment by AndSaf1** [comment]
Sorry. It is too difficult for me. I can't understand the algorithms

**Comment by AndSaf1** [comment]
What do you advise to read ...

**Comment by alexgonor** [comment]
The most suitable that I found for this task: https://rubyrush.ru/steps/loops.html
But the main idea for this task create logical algorithm how it could be done and decompose task to many small tasks and solving these small tasks, for example:
1. Take string of brackets and separate it to array
2. We know that it should be open bracket at the start and closed at the end. So, we can take first and last array elements and compare them, if they equal then we can took next pair after first and last.
3. And so on...

**Pull request merged**